libraries/botbuilder/src/teamsActivityHandler.ts

```diff
@@ -0,0 +1,498 @@
+        const runDialogs = async (): Promise<void> => {
+            await this.handle(context, 'Dialog', async () => {
+                // noop
+            });
+        };
+
+        // List of all Activity Types:
+        // https://github.com/Microsoft/botbuilder-js/blob/master/libraries/botframework-schema/src/index.ts#L1627
+        await this.handle(context, 'Turn', async () => {
+            switch (context.activity.type) {
+                case ActivityTypes.Message:
+                    await this.handle(context, 'Message', runDialogs);
+                    break;
+                case ActivityTypes.ConversationUpdate:
+                    await this.handle(context, 'ConversationUpdate', async () => {
+                        if (context.activity.membersAdded && context.activity.membersAdded.length > 0) {
+                            await this.handle(context, 'MembersAdded', runDialogs);
+                        } else if (context.activity.membersRemoved && context.activity.membersRemoved.length > 0) {
+                            await this.handle(context, 'MembersRemoved', runDialogs);
+                        } else {
+                            await runDialogs();
+                        }
+                    });
+                    break;
+                case ActivityTypes.MessageReaction:
+                    await this.handle(context, 'MessageReaction', async () => {
+                        if (context.activity.reactionsAdded && context.activity.reactionsAdded.length > 0) {
+                            await this.handle(context, 'ReactionsAdded', runDialogs);
+                        } else if (context.activity.reactionsRemoved && context.activity.reactionsRemoved.length > 0) {
+                            await this.handle(context, 'ReactionsRemoved', runDialogs);
+                        } else {
+                            await runDialogs();
+                        }
+                    });
+                    break;
+                case ActivityTypes.Event:
+                    await this.handle(context, 'Event', async () => {
+                        if (context.activity.name === 'tokens/response') {
+                            await this.handle(context, 'TokenResponseEvent', runDialogs);
+                        } else {
+                            await runDialogs();
+                        }
+                    });
+                    break;
+                case ActivityTypes.Invoke:
+                    // The onInvokeActivity handler should not call `runDialogs`, it needs to let the
+                    // typed-Invoke Activity handler call `runDialogs`.
+                    await this.handle(context, 'Invoke', async () => { });
+                    let invokeResponse: InvokeResponse;
+                    switch (context.activity.name) {
+                        case 'fileConsent/invoke':
+                            await this.handle(context, 'TeamsFileConsent', async () => { });
```

Comment:
can we make the default behavior run the child events... i.e. make the "on" an assignment rather than an add to a list